### ISSUE TITLE
tg unable to msg users before it was messaged first

### ISSUE THREAD

**Issue by SyRenity** [vysheng/tg#483]
Since February, tg is unable to send to any user, unless this user has contacted the tg user first.
Doing msg id#UID does nothing.

There was a work-around to load all contacts via contact_list, however it stopped working.
The other work-around is dialog_list that allows loading latest dialogs, however it does not cover all of connected users.

**Comment by LukeLR** [comment]
I think I have the same bug when sending messages by lua script, if the client didn't receive a message from that contact in the current session, you can't send a message to that chat.

**Comment by SyRenity** [comment]
@LukeLR The work around I use (as mentioned in #484) is to type dialog_list (which unfortunately doesn't work from LUA either now).

The big bummer is that all worked perfectly just 2 months ago, seems as Telegram changed the protocol.

Let's hope the @vysheng will resolve this in the upcoming #476

**Comment by LukeLR** [comment]
Yes, as in #480 dialog_list returns false for me when called from lua script… I wanted to workaround the sending bug by this as well…

**Comment by Requilence** [comment]
You need to use correct peer name **msg user#id** or **msg chat#id**

**Comment by LukeLR** [comment]
So, what's the right format to specify the user I want to get the history of? May I just pass the ID as an Integer, or the screen name as a String?

**Comment by SyRenity** [comment]
@Requilence I've tried it and it doesn't work - only the users that contacted Telegram CLI can be messaged.

**Comment by LukeLR** [comment]
Hey, I got the Trick! I am using the latest Version of tg-cli from the Master-branch (not the test-branch!) and tried to pass command line option -W when launching the daemon. -W means that tg-cli first executes a "dialog_list" before accepting any input. Seems useful! As a result, telegram-cli can now message users even without them to have tg-cli messaged first. I could swear that I have tried this when the problem occurred to me first with no result, but now it works! You should try and report if it works for you, too!

**Comment by tpmediarpi** [comment]
Thanks! Works fine for me!

**Comment by LukeLR** [comment]
you're welcome! I think I'm as glad as you are ;) But this should not be closed, since it is an issue and needs to be fixed some time...

**Comment by zeandrade** [comment]
@LukeLR perfect.I used -W param., and cli works fine! Without -W param, if you execute contact_list first, you can send messages too. ( I'm using  1.2.0 version)

**Comment by luckydonald** [comment]
Does this only apply to known users?
For example, If i switch the account (phone number) and use ```chat#1234``` from a database?

**Comment by LukeLR** [comment]
You mean, if the Chat was Never accessed on the client before?

**Comment by luckydonald** [comment]
Exactly. 
Like if I give you my user id, like  ```user#50441674```, and you don't have that contact in telegram.

**Comment by LukeLR** [comment]
I’ll guess that it wont work for that, because I think that the problem is that the client needs the chat to be cached locally for sending a message to it. therefore this can be fixed by loading the dialog list and therefore fetching dialogs from the server. But if you have never chatted with that person, there won’t be a dialog included in dialog list and therefore this fix won’t work. But I have not tried it yet, my cli only messages me or others that sent a message as a reply, so I did not need to message someone unknown. Maybe you should try by yourself? I think vysheng has published his contact data somewhere for questions, feature requests and so on, you could try to message him if you haven’t before ;)
Lukas Rose
Ingeborg-Bachmann-Straße 17
47506 Neukirchen-Vluyn
Telefon: 02845-949685
Mobil: 01578-1664674
E-Mail: ichkilldichgleichduzockeropfer@googlemail.com
Alternativ: lukas.rose@gmx.de

**Comment by luckydonald** [comment]
For your information, you appended your address to that email- not sure if wanted.

**Comment by LukeLR** [comment]
Thanks, removed it! Automatic signature of my email-client, should have known better... ;)
